### PR TITLE
Improve integration test suite CI run script

### PR DESCRIPTION
**Pull request by eseliger**
**The final result in the CI logs** 

image

Otherwise this PR does the following improvements:
- Fixes the CSS modules types generator when run outside of the repo root (`client/web`, for example)
- Fixes a deprecation in the Webpack dev server, which I saw along the way
- Adds the CSS generator to the set of generators run for build-web. They were missing there
- Share the set of generators to run for build-web, so this issue doesn't arise again
- Switch to a different run script where `yarn generate` isn't run twice
- The above grouping of log outputs into "steps"

### CODE REVIEW

File: client/shared/dev/generateCssModulesTypes.js [hunk 1]
```diff
@@ -14,6 +15,7 @@ function cssModulesTypings() {
   return spawn(BIN, ARGS, {
     stdio: 'inherit',
     shell: true,
+    cwd: REPO_ROOT,
```
Comment (eseliger) [inline client/shared/dev/generateCssModulesTypes.js:18]: Fixes the generator when run from a different cwd, such as `client/web`.

File: client/web/gulpfile.js [hunk 2]
```diff
@@ -128,19 +130,17 @@ async function webpackDevelopmentServer() {
   }
 
   const server = new WebpackDevServer(options, createWebpackCompiler(webpackConfig))
-  await new Promise((resolve, reject) => {
-    signale.await('Waiting for Webpack to compile assets')
-    server.listen(3080, '0.0.0.0', error => (error ? reject(error) : resolve()))
-  })
+  signale.await('Waiting for Webpack to compile assets')
+  await server.start()
```
Comment (eseliger) [inline client/web/gulpfile.js:134]: Deprecated.

File: client/web/gulpfile.js [hunk 3]
```diff
@@ -128,19 +130,17 @@ async function webpackDevelopmentServer() {
...
 }
 
 const developmentServer = DEV_WEB_BUILDER === 'webpack' ? webpackDevelopmentServer : esbuildDevelopmentServer
 
 // Ensure the typings that TypeScript depends on are build to avoid first-time-run errors
-const codeGen = gulp.parallel(schema, graphQlOperations, graphQlSchema)
+const codeGen = gulp.parallel(schema, graphQlSchema, graphQlOperations, cssModulesTypings)
```
Comment (eseliger) [inline client/web/gulpfile.js:140]: Were missing here.

File: client/web/gulpfile.js [hunk 4]
```diff
@@ -128,19 +130,17 @@ async function webpackDevelopmentServer() {
...
 
 // Watches code generation only, rebuilds on file changes
-const watchCodeGen = gulp.parallel(watchSchema, watchGraphQlSchema, watchGraphQlOperations)
+const watchCodeGen = gulp.parallel(watchSchema, watchGraphQlSchema, watchGraphQlOperations, watchCSSModulesTypings)
```
Comment (eseliger) [inline client/web/gulpfile.js:143]: Were missing here.

File: gulpfile.js [hunk 5]
```diff
@@ -7,21 +7,13 @@ const {
...
-/**
- * Starts all watchers on schema files.
- */
-const watchGenerators = gulp.parallel(watchSchema, watchGraphQlSchema, watchGraphQlOperations, watchCSSModulesTypings)
+const {
+  webpack: webWebpack,
+  developmentServer,
+  codeGen: generate,
+  watchCodeGen: watchGenerators,
+} = require('./client/web/gulpfile')
```
Comment (eseliger) [inline gulpfile.js:16]: Reusing them, so we don't run into this again :)

File: enterprise/dev/ci/internal/ci/pipeline-steps.go [hunk 6]
```diff
@@ -100,11 +100,11 @@ func addSharedTests(c Config) func(pipeline *bk.Pipeline) {
 			bk.Env("PUPPETEER_SKIP_CHROMIUM_DOWNLOAD", "true"), // Don't download browser, we use "download-puppeteer-browser" script instead
 			bk.Env("ENTERPRISE", "1"),
 			bk.Env("PERCY_ON", "true"),
-			bk.Cmd("COVERAGE_INSTRUMENT=true dev/ci/yarn-run.sh build-web"),
-			bk.Cmd("yarn --cwd client/shared run download-puppeteer-browser"),
-			bk.Cmd("yarn percy exec -- yarn run cover-integration"),
-			bk.Cmd("yarn nyc report -r json"),
-			bk.Cmd("dev/ci/codecov.sh -c -F typescript -F integration"),
+			bk.Cmd("COVERAGE_INSTRUMENT=true dev/ci/yarn-build.sh client/web"),
```
Comment (eseliger) [inline enterprise/dev/ci/internal/ci/pipeline-steps.go:103]: Prevents running yarn generate twice.

File: enterprise/dev/ci/internal/ci/pipeline-steps.go [hunk 7]
```diff
@@ -100,11 +100,11 @@ func addSharedTests(c Config) func(pipeline *bk.Pipeline) {
...
+			bk.Cmd("echo \"--- Install puppeteer\" && yarn --cwd client/shared run download-puppeteer-browser"),
+			bk.Cmd("echo \"--- Run integration test suite\" && yarn percy exec -- yarn run cover-integration"),
+			bk.Cmd("echo \"--- Process NYC report\" && yarn nyc report -r json"),
+			bk.Cmd("echo \"--- Upload coverage report\" && dev/ci/codecov.sh -c -F typescript -F integration"),
```
Comment (eseliger) [inline enterprise/dev/ci/internal/ci/pipeline-steps.go:107]: See PR description for how this looks in CI.

File: enterprise/dev/ci/internal/ci/pipeline-steps.go (hunk 6 above)
Comment (valerybugakov) [reply enterprise/dev/ci/internal/ci/pipeline-steps.go:103]: Nice, I noticed this issue too. Thanks for improving it! 🎉

File: client/web/gulpfile.js (hunk 2 above)
Comment (valerybugakov) [reply client/web/gulpfile.js:134]: Could you add the same change to the `client/web/dev/server/development.server.ts`? 🤗

File: client/web/gulpfile.js (hunk 3 above)
Comment (valerybugakov) [reply client/web/gulpfile.js:140]: Nice!

File: gulpfile.js [hunk 8]
```diff
@@ -7,21 +7,13 @@ const {
...
-
-/**
- * Starts all watchers on schema files.
- */
-const watchGenerators = gulp.parallel(watchSchema, watchGraphQlSchema, watchGraphQlOperations, watchCSSModulesTypings)
+const {
+  webpack: webWebpack,
+  developmentServer,
+  codeGen: generate,
+  watchCodeGen: watchGenerators,
```
Comment (valerybugakov) [inline gulpfile.js:15]: optional: it would be great to rename them to make naming consistent through all gulp files

**Review by valerybugakov — APPROVED** [review]
That's great! Thanks for taking the time to improve this 💪

File: client/web/gulpfile.js (hunk 2 above)
Comment (eseliger) [reply client/web/gulpfile.js:134]: Will do!

**Pull request merged**